### PR TITLE
Update StorageEngine and Spark connectors design doc

### PR DESCRIPTION
**Pull request by SilverNarcissus**
Update chinese design doc about spark and data manipulation

### CODE REVIEW

File: docs/Documentation-CHN/SystemDesign/7-Connector/4-Spark-IOTDB.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+|    1 | null                          | true                     | null                       | 2.2                           | true                     | null                       |
+|    2 | null                          | false                    | aaa                        | 2.2                           | null                     | null                       |
+|    3 | null                          | null                     | null                       | 2.1                           | true                     | null                       |
+|    4 | null                          | true                     | bbb                        | null                          | null                     | null                       |
+|    5 | null                          | null                     | null                       | null                          | false                    | null                       |
+|    6 | null                          | null                     | ccc                        | null                          | null                     | null                       |
+
+窄表结构: 关系型数据库模式，IOTDB group by device格式
```
Comment (Alima777) [inline docs/Documentation-CHN/SystemDesign/7-Connector/4-Spark-IOTDB.md]: It's renamed as "align by device" now.

File: docs/Documentation-CHN/SystemDesign/7-Connector/4-Spark-IOTDB.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+|    2 | root.ln.wf02.wt01             | null                     | null                       | 2.2                          |                 
+|    2 | root.ln.wf02.wt02             | false                    | aaa                        | null                           |                   
+|    3 | root.ln.wf02.wt01             | true                     | null                       | 2.1                           |                 
+|    4 | root.ln.wf02.wt02             | true                     | bbb                        | null                          |                  
+|    5 | root.ln.wf02.wt01             | false                    | null                       | null                          |                   
+|    6 | root.ln.wf02.wt02             | null                     | ccc                        | null                          |                   
+
+由于IOTDB查询到的数据默认为宽表结构，所以需要宽窄表转换，有如下两个实现方法
+
+#### 1. 使用IOTDB的group by device语句
```
Comment (Alima777) [inline docs/Documentation-CHN/SystemDesign/7-Connector/4-Spark-IOTDB.md]: The same as above.

File: docs/Documentation-CHN/SystemDesign/7-Connector/4-Spark-IOTDB.md (hunk 1 above)
Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/7-Connector/4-Spark-IOTDB.md]: fixed~

File: docs/Documentation-CHN/SystemDesign/7-Connector/4-Spark-IOTDB.md (hunk 2 above)
Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/7-Connector/4-Spark-IOTDB.md]: fixed~

File: docs/Documentation-CHN/SystemDesign/7-Connector/3-Spark-TsFile.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+#### 5. 查询实际执行
+实际数据查询执行由Tsfile原生组件完成，参见：
+* [Tsfile原生查询流程](../1-TsFile/4-Read.md)
+
+## 写入步骤流程
+写入主要是将Dataframe结构中的数据转化为Tsfile的RowRecord，使用Tsfile Writer进行写入
+
+#### 宽表结构
+其主要转化代码在如下两个文件中：
+* src/main/scala/org/apache/iotdb/spark/tsfile/WideConverter.scala 负责结构转化
```
Comment (qiaojialin) [inline docs/Documentation-CHN/SystemDesign/7-Connector/3-Spark-TsFile.md]: add a blank line before all *

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+# 数据分区
+
+时间序列数据在存储组和时间范围两个层级上进行分区
+
+## 存储组
+
+存储组由用户显示指定，使用语句"SET STORAGE GROUP TO"来指定存储组，每一个存储组有一个对应的StoragetGroupProcessor
+
+### 相关代码
+* src/main/java/org/apache/iotdb/db/engine/StorageEngine.java
```
[the remark this answers is not in the document]
Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md]: fixed~

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+
+## 时间范围
+
+同一个存储组中的数据按照用户指定的时间范围进行分区，相关参数为partition_interval，默认为周，也就是不同周的数据会放在不同的分区中
+
+### 实现逻辑
+
+StoragetGroupProcessor对插入的数据进行分区计算，找到指定的TsfileProcessor，而每一个TsfileProcessor对应的tsfile会被放置在不同的分区文件夹内
```
[the remark this answers is not in the document]
Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md]: fixed~

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+------ [时间分区ID2]
+
+---- [存储组名2]
+
+-- unsequence
+
+### 相关代码
+
+* src/main/java/org/apache/iotdb/db/engine/storagegroup.StoragetGroupProcessor.java中的getOrCreateTsFileProcessorIntern方法
```
[the remark this answers is not in the document]
Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md]: fixed~

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/6-DataManipulation.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+# 数据操控设计简介
```
[the remark this answers is not in the document]
Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/4-StorageEngine/6-DataManipulation.md]: fixed~

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+-->
+
+# 数据分区
+
+时间序列数据在存储组和时间范围两个层级上进行分区
```
[the remark this answers is not in the document]
Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md]: fixed~

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/6-DataManipulation.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+
+## 数据TTL设置
+
+* 对应的接口
+	* JDBC 的 execute 接口，使用SET TTL语句
+
+* 总入口: public void setTTL(String storageGroup, long dataTTL) StorageEngine.java
+    * 找到对应的 StorageGroupProcessor
+    * 设置新的data ttl
```
[the remark this answers is not in the document]
Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/4-StorageEngine/6-DataManipulation.md]: fixed~

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/6-DataManipulation.md [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+* 总入口: public void delete(String deviceId, String measurementId, long timestamp) StorageEngine.java
+    * 找到对应的 StorageGroupProcessor
+    * 找到受影响的所有TsfileProcessor
+    * 写写前日志
+    * 找到受影响的所有TsfileResource
+    * 在mod文件中记录删除的时间点
+    * 如果文件没有关闭（存在对应的TsfileProcessor），则删除内存中的数据
+
+
+## 数据TTL设置
```
[the remark this answers is not in the document]
Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/4-StorageEngine/6-DataManipulation.md]: fixed~

File: docs/Documentation-CHN/SystemDesign/7-Connector/3-Spark-TsFile.md (hunk 3 above)
Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/7-Connector/3-Spark-TsFile.md]: fixed~

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+
+* 每个时间分区所对应的未关闭的乱序文件列表
+
+* 该存储组的全部顺序文件列表（按照时间排序）
+
+* 该存储组的全部乱序文件列表（无顺序）
+
+* 记录每一个设备最后写入时间的map
+
+* 记录每一个设备最后刷盘时间的map
```
Comment (qiaojialin) [inline docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md]: 加上字段名字吧。这个加上 是干什么用的 ：用来区分顺序和乱序数据

Comment (qiaojialin) [reply docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md]: 以后用 apache 的仓库提交分支吧，我就可以直接改了

Comment (SilverNarcissus) [reply docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md]: OK. I will fix this~

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md [hunk 12]
```diff
@@ -0,0 +1,86 @@
...
+
+-->
+
+# 数据分区
+
+时间序列数据在存储组和时间范围两个层级上进行分区
+
+## 存储组
+
+存储组由用户显示指定，使用语句"SET STORAGE GROUP TO"来指定存储组，每一个存储组有一个对应的StoragetGroupProcessor
```
Comment (qiaojialin) [inline docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md]:
```suggestion
存储组由用户显示指定，使用语句"SET STORAGE GROUP TO"来指定存储组，每一个存储组有一个对应的 StorageGroupProcessor
```

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md [hunk 13]
```diff
@@ -0,0 +1,86 @@
...
+
+其拥有的主要字段为：
+
+* 一个读写锁: insertLock
+
+* 每个时间分区所对应的未关闭的顺序文件树: workSequenceTsFileProcessors
```
Comment (qiaojialin) [inline docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md]:
```suggestion
* 每个时间分区所对应的未关闭的顺序文件处理器: workSequenceTsFileProcessors
```

File: docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md [hunk 14]
```diff
@@ -0,0 +1,86 @@
...
+
+* 每个时间分区所对应的未关闭的乱序文件列表: workUnsequenceTsFileProcessors
```
Comment (qiaojialin) [inline docs/Documentation-CHN/SystemDesign/4-StorageEngine/5-DataPartition.md]:
```suggestion
* 每个时间分区所对应的未关闭的乱序文件处理器: workUnsequenceTsFileProcessors
```

**Pull request merged**